sw/airborne/modules/computer_vision/textons.c

```diff
@@ -225,6 +316,25 @@ void DictionaryTrainingYUV(uint8_t *frame, uint16_t width, uint16_t height)
       }
     }
 
+    // make sure that the other threads have access to the normalized distribution
+    // and not the one we are going to fill / change.
+    pthread_mutex_lock(&textons_mutex);
```

Comment:
Let's indeed meet. One of the arrays is continuously being filled, while the other one has the result of the previous calculations for the previous image. External files only have a pointer to the finished distribution, not the one being filled. The only point where this could still go wrong is at the moment of the switch, where I switch the pointer to the array that is going to be filled for the next image. I think it would be strange to leave this to the "reading side", as I don't know up front which different files will want to read it in the future. Also, the time of the lock is truly minimal like this (in other cases I have seen copying of information between locking and unlocking, here it is just the switch of a pointer). But it could be that I misunderstand what a mutex does.